site-manage/site-manage-util/util/src/java/org/sakaiproject/site/util/Participant.java

```diff
@@ -176,21 +176,20 @@ public int compareTo(Participant participantB) {
 
 		try {
 			userA = UserDirectoryService.getUser(uniqname);
```

Comment:
I feel that if we performed a pre comparison on uniquename could avoid some unnecessary calls, i.e.
```
if (uniqnameA == null || uniqnameB == null) {
    return uniqnameA == uniqnameB ? 0 : (uniqnameA == null ? -1 : 1);
}
```
this pre check would handle the case if any were null without instantiating any objects. Should only compare objects when both uniquename are not null. A further improvement would be both to evaluate if uniquename are equal.